### ISSUE TITLE
bug in coarse channel frequency calculation

### ISSUE THREAD

**Issue by lacker** [UCBerkeleySETI/hpguppi_proc#5]
I think there's a bug in the code that calculates the center frequencies of the coarse channels - https://github.com/UCBerkeleySETI/hpguppi_proc/blob/mark_r/src/hpguppi_coherent_bf_thread.c#L384

Basically this expression:

```
((n_chan_per_node-1)/2)
```

since it's an integer expression, it's doing integer arithmetic, which means truncating the result of a division. I think the intention of this code is to calculate the center of the channel and in that case it should be doing a floating point division.

There's a similar issue in the upchannelizing version of the code and perhaps elsewhere.

**Comment by david-macmahon** [comment]
Yes, I think `nchan_chan_per_node` should be cast to a `double` like this:
```c
((((double)nchan_per_node)-1) / 2)
```
or change `1` to `1.0` like this:
```c
((nchan_per_node-1.0)/2)
```

**Comment by mruzinda** [comment]
Thanks for pointing this out. I'll make that change in the next commit.
